src/main/java/com/glodblock/github/network/CPacketRenamer.java

```diff
@@ -97,15 +99,23 @@ public void toBytes(ByteBuf buf) {
 
     public static class Handler implements IMessageHandler<CPacketRenamer, IMessage> {
 
+        private String getName(ICustomNameObject obj) {
+            // "‎" is a zero size character, sent because simple channel does not support empty string
```

Comment:
I am not sure about this change in general actually. SimpleChannel doesn't have anything to do with what strings it supports; that is up to the implementation of CPacketRenamer.

What is the crash message? I can't reproduce it in dev.

Reply:
You need to use the rename in wireless pattern terminal on a crafting input bus, if this patch is not installed it will result in 

```
SimpleChannelHandlerWrapper exception
java.lang.IllegalArgumentException: Empty string not allowed
	at net.minecraft.nbt.NBTTagString.<init>(NBTTagString.java:23) ~[dx.class:?]
	at net.minecraft.nbt.NBTTagCompound.func_74778_a(NBTTagCompound.java:105) ~[dh.class:?]
	at com.glodblock.github.util.Util$DimensionalCoordSide.writeToNBT(Util.java:611) ~[Util$DimensionalCoordSide.class:?]
	at com.glodblock.github.inventory.item.WirelessInterfaceTerminalInventory.writeToNBT(WirelessInterfaceTerminalInventory.java:55) ~[WirelessInterfaceTerminalInventory.class:?]
	at com.glodblock.github.inventory.item.WirelessInterfaceTerminalInventory.setClickedInterface(WirelessInterfaceTerminalInventory.java:138) ~[WirelessInterfaceTerminalInventory.class:?]
	at com.glodblock.github.network.CPacketRenamer$Handler.onMessage(CPacketRenamer.java:134) ~[CPacketRenamer$Handler.class:?]
	at com.glodblock.github.network.CPacketRenamer$Handler.onMessage(CPacketRenamer.java:98) ~[CPacketRenamer$Handler.class:?]
	at cpw.mods.fml.common.network.simpleimpl.SimpleChannelHandlerWrapper.channelRead0(SimpleChannelHandlerWrapper.java:37) ~[SimpleChannelHandlerWrapper.class:?]
	at cpw.mods.fml.common.network.simpleimpl.SimpleChannelHandlerWrapper.channelRead0(SimpleChannelHandlerWrapper.java:17) ~[SimpleChannelHandlerWrapper.class:?]
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:98) ~[SimpleChannelInboundHandler.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:337) [DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:323) [DefaultChannelHandlerContext.class:?]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) [MessageToMessageDecoder.class:?]
	at io.netty.handler.codec.MessageToMessageCodec.channelRead(MessageToMessageCodec.java:111) [MessageToMessageCodec.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:337) [DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:323) [DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:785) [DefaultChannelPipeline.class:?]
	at io.netty.channel.embedded.EmbeddedChannel.writeInbound(EmbeddedChannel.java:169) [EmbeddedChannel.class:?]
	at cpw.mods.fml.common.network.internal.FMLProxyPacket.func_148833_a(FMLProxyPacket.java:77) [FMLProxyPacket.class:?]
	at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:212) [ej.class:?]
	at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:165) [nc.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:659) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:547) [MinecraftServer.class:?]
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:186) [bsx.class:?]
	at fastcraft.u.a(F:289) [fastcraft-1.25.jar:?]
	at fastcraft.H.aq(F:36) [fastcraft-1.25.jar:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:396) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer$2.run(MinecraftServer.java:685) [?:?]
```

Reply:
it's more like nbt in general, null is not allowed, and custom name start as null. So I guess just checking `hasCustomName` can already solve the issue, no need the empty charecter.